tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_power_iterative.h

```diff
@@ -13,24 +13,264 @@ using namespace sfpi;
+    // N_m = 2**23
+    // bias = 0x3f800000
+
+    // In this case, we transform the formula to:
+    // z = (bias) * N_m + (x * log2(a)) * N_m
+    // where (bias + N_m) = 0x3f800000
+    // and (x * log2(a)) * N_m = addexp(z_f32, 23)
+
+    // Notes:
+    // - N_m being a power of 2 ensures equivalent results
+    // - addexp(z_f32, 23) is used because it translates to a single-cycle SFPDIVP2
+    //   instruction with immediate operand (i.e. no extra register used).
+    //   (vs. 1 cycle SFPLOADI + 2 cycles MAD)
+
+    z_f32 = addexp(z_f32, 23);  // equal to multiplying by 2**23
+    const sfpi::vFloat bias = sfpi::vFloat(0x3f800000);
+    sfpi::vInt z = _float_to_int32_positive_(z_f32 + bias);
+
+    sfpi::vInt zii = exexp(sfpi::reinterpret<sfpi::vFloat>(z));         // Note: z & 0x7f800000 in paper
+    sfpi::vInt zif = sfpi::exman9(sfpi::reinterpret<sfpi::vFloat>(z));  // Note: z & 0x007fffff in paper
+
+    // Compute formula in Horner form
+    sfpi::vFloat d1 = sfpi::vFloat(0.40196114e-7);
+    sfpi::vFloat d2 = sfpi::int32_to_float(sfpi::vInt(0xf94ee7) + zif, 0);
+    sfpi::vFloat d3 = sfpi::int32_to_float(sfpi::vInt(0x560e) + zif, 0);
+
+    d2 = d1 * d2;
+    zif = _float_to_int32_positive_(d2 * d3);
+
+    // Restore exponent
+    zii = sfpi::reinterpret<sfpi::vInt>(sfpi::setexp(sfpi::reinterpret<sfpi::vFloat>(zif), 127U + zii));
+
+    sfpi::vFloat y = sfpi::reinterpret<sfpi::vFloat>(zii);
+
+    // Post-processing: ensure that special values (e.g. 0**0, -1**0.5, ...) are handled correctly
+    // Check valid base range
+    sfpi::vInt pow_int =
+        sfpi::float_to_int16(pow, 0);  // int16 should be plenty, since large powers will approach 0/Inf
+    sfpi::vFloat pow_rounded = sfpi::int32_to_float(pow_int, 0);
+
+    // Division by 0 when base is 0 and pow is negative => set to NaN
+    v_if((abs_base == 0.f) && pow < 0.f) {
+        y = sfpi::vConstFloatPrgm2;  // negative powers of 0 are NaN, e.g. pow(0, -1.5)
+    }
+    v_endif;
+
+    // Negative base handling (for both positive and negative exponents)
+    v_if(base < 0.0f) {
+        // If pow is odd integer then result is negative
+        // If power is even, then result is positive
+        // To get the sign bit of result, we can shift last bit of pow_int to the 1st bit
+        y = setsgn(y, pow_int << 31);
+
+        // Check for integer power, if it is not then overwrite result with NaN
+        v_if(pow_rounded != pow) {  // negative base and non-integer power => set to NaN
+            y = sfpi::vConstFloatPrgm2;
+        }
+        v_endif;
+    }
+    v_endif;
+
+    // LRegs work on float32 data. If DST is bfloat16 then SFPSTORE will truncate it.
+    // This can reduce accuracy: for instance, 9**2 = 80.8 gets round to 80.5
+    // rather than 81 (which would have been correct).
+    // To avoid this issue, we explicitly convert to bfloat16 using round-to-nearest-even.
+    y = reinterpret<sfpi::vFloat>(sfpi::float_to_fp16b(y, 0));
```

Comment:
```suggestion
    if constexpr (!is_fp32_dest_acc_en) {
        // LRegs work on float32 data. If DST is bfloat16 then SFPSTORE will truncate it.
        // This can reduce accuracy: for instance, 9**2 = 80.8 gets round to 80.5
        // rather than 81 (which would have been correct).
        // To avoid this issue, we explicitly convert to bfloat16 using round-to-nearest-even.
        y = reinterpret<sfpi::vFloat>(sfpi::float_to_fp16b(y, 0));
    }
```
(we don't want to round to bf16 If data is fp32)